Basic/2DSpaceShooter/Assets/Scripts/RandomPositionPlayerSpawner.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using MLAPI;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+[RequireComponent(typeof(NetworkManager))]
+public class RandomPositionPlayerSpawner: MonoBehaviour
+{
+    NetworkManager m_NetworkManager;
+
+    int m_RoundRobinIndex = 0;
+    
+    [SerializeField]
+    SpawnMethod m_SpawnMethod;
+    
+    [SerializeField]
+    List<Vector3> m_SpawnPositions = new List<Vector3>() { Vector3.zero };
+
+    /// <summary>
+    /// Get a spawn position for a spawned object based on the spawn method.
+    /// </summary>
+    /// <returns>?The spawn position.</returns>
+    /// <exception cref="NotImplementedException"></exception>
+    public Vector3 GetNextSpawnPosition()
```

Comment:
Using the RoundRobin method, one potential issue using a Getter this way that modifies a member variable is if you call it twice GetNextSpawnposition()  it will return 2 different values, and someone that will get this script may hunt for hours to debug the issue.

Reply:
We could add it a comment to pay attention to use the cache value of the GetNextSpawnPosition()

Reply:
That's why I made it a function instead of a property. I'd argue that calling a function which is called `GetNext` should come with the expectation that it will change the the internal state of this.

Reply:
Same as `Random.NextDouble` etc. basically.

Reply:
I really hope that.